src/obs_inv_utils/aws_s3_interface.py

```diff
@@ -0,0 +1,243 @@
+import re
+import attr
+from collections import namedtuple, OrderedDict
+import boto3
+from datetime import datetime
+from botocore.config import Config
+from botocore import UNSIGNED
+
+
+bdp_config = Config(
+    region_name = 'us-east-1',
+    signature_version = UNSIGNED,
+    retries = {
+        'max_attempts': 10,
+        'mode': 'standard'
+    }
+)
+
+nl = '\n'
+
+AwsS3ObjectsListContents = namedtuple(
+    'AwsS3ObjectsListContents',
+    [
+        'prefix',
+        'files_count',
+        'listed_objects',
+        'obs_cycle_time',
+        'submitted_at',
+        'latency'
+    ],
+)
+
+AwsS3CommandRawResponse = namedtuple(
+    'AwsS3CommandRawResponse',
+    [
+        'command',
+        'return_code',
+        'output',
+        'success',
+        'args_0',
+        'submitted_at',
+        'latency'
+    ],
+)
+
+
+AwsS3FileMeta = namedtuple(
+    'AwsS3FileMeta',
+    [
+        'name',
+        'permissions',
+        'last_modified',
+        'size'
+    ],
+)
+
+
+AwsS3Command = namedtuple(
+    'AwsS3Command',
+    [
+        'command',
+        'arg_validator',
+        'output_parser'
+    ],
+)
+
+AWS_BDP_BUCKET = 'noaa-reanalyses-pds'
+CMD_GET_S3_OBJ_LIST = 'list_objects'
+
+
+def get_bdp_s3_client():
```

Comment:
will your code also work for a private bucket or just for bdp?

Reply:
For now, it only works for the BDP bucket, but can easily be modified to allow private buckets.